dashboard/test/lib/services/script_seed_test.rb

```diff
@@ -314,9 +314,11 @@ class ScriptSeedTest < ActiveSupport::TestCase
       script_with_changes, json = get_script_and_json_with_change_and_rollback(script) do
         lesson = script.lessons.first
         lesson.vocabularies.first.update!(definition: 'updated definition')
+        key = Vocabulary.sanitize_key("#{lesson.name}-vocab-3")
+        key = Vocabulary.uniquify_key(key, script.course_version.id)
         lesson.vocabularies.create(
           word: 'new word',
-          key: "#{lesson.name}-vocab-3",
+          key: key,
```

Comment:
it seems like using a factory to create the vocab could avoid the need for the uniquify_key method. Not suggesting you do that now, but wondering if you agree with this assessment @Hamms so that we have some ideas for how to clean this up when we have time after April deadline.

Reply:
Theoretically, but I found that when I tried to specify a word here and let the key be generated a whole bunch of stuff in this test case broke in ways that were not obvious to me. Hence my complaints about this test case :)

Reply:
Ah, this was failing because numbers are now disallowed right? I'm not sure what the best solution is for that, other than changing the numbers in the lesson names to letters or something

Reply:
If it solves a problem, allowing numbers in vocab keys seems ok to me.

Reply:
note that this test is not actually the one for which uniqueness is necessary; I just threw that in for consistency. It's the diff below - the one in `create_script_tree` - that's the real problem

Reply:
Ah good point, though I think the issue is the same (using numbers). I agree this is out of the scope of this PR. As long as `uniquify_key` is deterministic across runs (which it is), I think this is a fine solution for now.